main/lsp/ExtractVariable.cc

```diff
@@ -13,66 +13,89 @@ void logDebugInfo(const std::shared_ptr<spdlog::logger> logger, const core::Glob
     logger->error("source=\"{}\"", absl::CEscape(selectionLoc.file().data(gs).source()));
 }
 
+optional<core::LocOffsets> detectCase(const ast::ExpressionPtr *whereToInsert, const core::LocOffsets target) {
+    if (auto send = ast::cast_tree<ast::Send>(*whereToInsert)) {
+        if (send->fun == core::Names::caseWhen()) {
+            int numPatterns = core::cast_type_nonnull<core::IntegerLiteralType>(
+                                  ast::cast_tree_nonnull<ast::Literal>(send->getPosArg(1)).value)
+                                  .value;
+            for (int i = numPatterns + 2; i < send->numPosArgs(); i++) {
+                auto *body = &send->getPosArg(i);
```

Comment:
```suggestion
                const auto &body = &send->getPosArg(i);
```

With adjustments for the changed type of `body` below.